itou/asp/models.py

```diff
@@ -0,0 +1,159 @@
+# from django.db import models
+from enum import Enum
+
+
+class LaneType(Enum):
+    """
+    Lane type
+
+    Import/translation of ASP ref file: ref_type_voie_v3.csv
+    """
+
+    AER = "Aérodrome"
+    AGL = "Agglomération"
+    AIRE = "Aire"
+    ALL = "Allée"
+    ACH = "Ancien chemin"
+    ART = "Ancienne route"
+    AV = "Avenue"
+    BEGI = "Beguinage"
+    BD = "Boulevard"
+    BRG = "Bourg"
+    CPG = "Camping"
+    CAR = "Carrefour"
+    CTRE = "Centre"
+    CCAL = "Centre commercial"
+    CHT = "Chateau"
+    CHS = "Chaussee"
+    CHEM = "Chemin"
+    CHV = "Chemin vicinal"
+    CITE = "Cité"
+    CLOS = "Clos"
+    CTR = "Contour"
+    COR = "Corniche"
+    COTE = "Coteaux"
+    COUR = "Cour"
+    CRS = "Cours"
+    DSC = "Descente"
+    DOM = "Domaine"
+    ECL = "Ecluse"
+    ESC = "Escalier"
+    ESPA = "Espace"
+    ESP = "Esplanade"
+    FG = "Faubourg"
+    FRM = "Ferme"
+    FON = "Fontaine"
+    GAL = "Galerie"
+    GARE = "Gare"
+    GBD = "Grand boulevard"
+    GPL = "Grande place"
+    GR = "Grande rue"
+    GRI = "Grille"
+    HAM = "Hameau"
+    IMM = "Immeuble(s)"
+    IMP = "Impasse"
+    JARD = "Jardin"
+    LD = "Lieu-dit"
+    LOT = "Lotissement"
+    MAIL = "Mail"
+    MAIS = "Maison"
+    MAS = "Mas"
+    MTE = "Montee"
+    PARC = "Parc"
+    PRV = "Parvis"
+    PAS = "Passage"
+    PLE = "Passerelle"
+    PCH = "Petit chemin"
+    PRT = "Petite route"
+    PTR = "Petite rue"
+    PL = "Place"
+    PTTE = "Placette"
+    PLN = "Plaine"
+    PLAN = "Plan"
+    PLT = "Plateau"
+    PONT = "Pont"
+    PORT = "Port"
+    PROM = "Promenade"
+    QUAI = "Quai"
+    QUAR = "Quartier"
+    RPE = "Rampe"
+    REMP = "Rempart"
+    RES = "Residence"
+    ROC = "Rocade"
+    RPT = "Rond-point"
+    RTD = "Rotonde"
+    RTE = "Route"
+    RUE = "Rue"
+    RLE = "Ruelle"
+    SEN = "Sente"
+    SENT = "Sentier"
+    SQ = "Square"
+    TPL = "Terre plein"
+    TRAV = "Traverse"
+    VEN = "Venelle"
+    VTE = "Vieille route"
+    VCHE = "Vieux chemin"
+    VILL = "Villa"
+    VLGE = "Village"
+    VOIE = "Voie"
+    ZONE = "Zone"
+    ZA = "Zone d'activite"
+    ZAC = "Zone d'amenagement concerte"
+    ZAD = "Zone d'amenagement differe"
+    ZI = "Zone industrielle"
+    ZUP = "Zone urbanisation prio"
+
+    @classmethod
+    def with_similar_name(cls, name, fmt=str.upper):
+        "Returns enum with similar name (fmt-fn wise)"
+        assert name
+        for elt in cls:
+            test = fmt(name)
+            if test == fmt(elt.name):
+                return elt
+
+        return None
+
+    @classmethod
+    def with_similar_value(cls, value, fmt=str.lower):
+        "Returns enum with a similar value (fmt-fn wise)"
+        assert value
+        revert_map = {fmt(lt.value): lt for lt in cls}
```

Comment:
Est-ce que tu t'attends à ce que `lt for lt in cls` soit une liste ici ? C'est pas clair dans le code, et si c'est ce que tu attends il y a une erreur.

Reply:
Non, je voulais bien un dictionnaire inversé de l'enum

Reply:
C'est bien ça que tu veux ?

```python
{'Agglomération': 'AGL', 'Aire': 'AIRE', 'Allée': 'ALL', 'Ancien chemin': 'ACH', 'Ancienne route': 'ART' …}
```

Tu peux l'avoir en une seule ligne, même un peu plus lisible peut-être ?

```python
return {v.value: k for k, v in LaneType.__members__.items()}
```

Reply:
Pas en une seule ligne en fait… sorry

Reply:
En fait, je n'ai pas utilisé `__members__` par fainéantise (et méconnaissance). 

Ca fonctionne aussi, sauf si tu as des aliases dans l'énumération (ils sont alors cachés):
https://docs.python.org/3/library/enum.html#iteration